src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py

```diff
@@ -596,3 +600,33 @@ def closure_to_tasklet_sdfg(
     for output in outputs:
         context.body.arrays[output.value.data].transient = False
     return context, outputs
+
+
+def lambda_to_tasklet_sdfg(
+    node: itir.StencilClosure,
+    offset_provider: dict[str, Any],
+    node_types: dict[int, next_typing.Type],
+) -> tuple[Context, Sequence[tuple[str, ValueExpr]], list[ValueExpr]]:
+    body = dace.SDFG("tasklet_toplevel")
+    state = body.add_state("tasklet_toplevel_entry")
+    symbol_map: dict[str, ValueExpr | IteratorExpr] = {}
+
+    context = Context(body, state, symbol_map)
+
+    def get_lambda() -> Lambda:
+        stencil_fobj = cast(FunCall, node.stencil)
+        return cast(Lambda, stencil_fobj.args[0])
```

Comment:
In theory, the callee could be a `Lambda` as well as a `SymRef` to a `FunctionDefinition`, so it's dangerous to cast it to a `Lambda` without checking. While as of of now probably all will be just lambdas, we should handle the different callables uniformly.

Reply:
However, this cast happens in a function for visiting lambda functions -- `lambda_to_tasklet_sdfg`. Is it OK if I just add an assert on the instance type?
```suggestion
        assert isinstance(stencil_fobj.args[0], Lambda)
        return stencil_fobj.args[0]
```

Reply:
I think I'd probably split this function (`lambda_to_tasklet_sdfg`) into two:
- A function that really just turns a lambda into a tasklet SDFG -- this one you could reuse with the parallel case
- A function that extracts the lambda function from the scan `StencilClosure`
(- Another function that extracts the lambda function/func def from the parallel `StencilClosure`)

The problem is that it does not simply convert a lambda to a tasklet SDFG, but rather it extracts the lambda from a scan stencil closure and then it converts it to an SDFG, which breaks the single responsibility principle and make the code not super reusable.

Reply:
@petiaccja I have tried to address your comment in the latest commit

Reply:
Yeah, that looks better to me. What do you think?

Reply:
I agree on the single responsibility principle, and I like that we are now using one common function `closure_to_tasklet_sdfg`. It would be even nicer if the visitor functions would have the same return signature, but unfortunately this is not the case: the node `visit` function returns `context, inner_inputs, inner_outputs` in case of lambda function, only `inner_outputs` in case of function call.